### PR TITLE
Move Texture Coordinates for Spheres from Ch 6 to Ch4 to define (u,v)

### PR DESCRIPTION
**Pull request by shaunplee**
Replaces #497 (I couldn't figure out how to change an existing pull request to target a different branch), addressing #496.

### CODE REVIEW

**Comment by shaunplee** [comment]
Also, lots of commits because I kept failing to see where my anti-social Emacs configuration was quietly deleting whitespace. Sorry.

File: books/RayTracingTheNextWeek.html [hunk 1]
```diff
@@ -1669,9 +1669,9 @@
 Image Texture Mapping
 ====================================================================================================
 
-We used the hitpoint p before to compute $(u,v)$ surface coordinates to index a procedure solid
-texture like marble. We can also read in an image and use the 2D $(u,v)$ texture coordinate to
-index into the image.
+From the hitpoint $\mathbf{P}$ before to compute the surface coordinates $(u,v)$. We then use these
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html:1672]: “From the hitpoint $\mathbf{P}$, we compute the surface coordinates $(u,v)$.”

File: books/RayTracingTheNextWeek.html [hunk 2]
```diff
@@ -957,6 +957,71 @@
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     [Listing [hit-record-uv]: <kbd>[hittable.h]</kbd> Adding U,V coordinates to the `hit_record`]
 
+We will also need to compute $(u,v)$ texture coordinates for hittables.
+
+Texture Coordinates for Spheres
```
Comment (hollasch) [inline books/RayTracingTheNextWeek.html:963]: Two blank lines before H2 headers.

Comment (shaunplee) [reply books/RayTracingTheNextWeek.html:963]: I think I'm misunderstanding--I'm seeing blank lines 961 and 962 before line 963: "Texture Coordinates for Spheres". How should this look?

**Comment by hollasch** [comment]
Hmmm. I've actually taken your changes, rebased + squashed them (as indicated by the autosquash markup). You can find the result in PR #518. I did this because I was having a hard time grokking the diff of the sum of all changes.

Now, however, I'm curious to see if your autosquash chain will actually do the right thing with a "rebase and merge" on PR approval, so I'll test it out.

**Comment by hollasch** [comment]
Ah, that's right, it had merge conflicts in the changelog (you had added your changelog entry to a prior release, as the new one didn't have a section yet). Oh well.

Can you please look at PR #518 and let me know if that looks good? Thanks.

**Comment by shaunplee** [comment]
Oops, I thought I had tried to add a new version 3.1.1 in the changelog. (Sorry, still learning this open source contribution process.) I think PR #518 looks good. Thanks for your patience and for sorting it out!

**Comment by hollasch** [comment]
Ok, it's in. Thanks @shaunplee for the work! We're all new to the open source process, and really, every project is different, so we're all learning.

**Pull request closed without merging**